app/questionnaire/questionnaire_store_updater.py

```diff
@@ -170,7 +166,7 @@ def capture_dependencies_for_list_change(self, list_name: str) -> None:
             list_name
         )
         section_ids.update(
-            self._schema.list_collector_section_ids_by_list_name[list_name]
+            self._schema.list_collector_section_ids_by_list_name.get(list_name, ())
```

Comment:
I suggest using a `set` default than a `tuple` for clarity, even if it functionally works.